paml/paml.ttl

```diff
@@ -355,32 +355,51 @@ paml:SampleArray rdf:type owl:Class ;
         rdfs:subClassOf paml:SampleCollection ,
             [ rdf:type owl:Restriction ;
               owl:onProperty paml:containerType ;
-              owl:allValuesFrom xsd:anyURI ; # replace with container ontology, when available
+              owl:allValuesFrom paml:ContainerSpec ;
               owl:minCardinality "1"^^xsd:nonNegativeInteger ; owl:maxCardinality "1"^^xsd:nonNegativeInteger ;
               rdfs:comment "Type of container used for storing the samples. The size and dimension may not match that of the array: it is up to execution to lay out the array in one or more containers"
             ] ,
             [ rdf:type owl:Restriction ;
               owl:onProperty paml:contents ;
-              owl:allValuesFrom xsd:string ; # need to replace this with an serializable array model
+              owl:allValuesFrom paml:PrimitiveArray ; # need to replace this with an serializable array model
               owl:minCardinality "1"^^xsd:nonNegativeInteger ; owl:maxCardinality "1"^^xsd:nonNegativeInteger ;
               rdfs:comment "N-dimensional array of URI for specification or nulls"
             ] .
 
-paml:containerType rdf:type owl:DatatypeProperty ;
+paml:containerType rdf:type owl:ObjectProperty ;
         rdfs:comment '''''' ;
               rdfs:domain paml:SampleArray ;
-              rdfs:range xsd:anyURI ;
+              rdfs:range paml:ContainerSpec ;
               rdfs:label "container_type" .
 
-paml:contents rdf:type owl:DatatypeProperty ;
+paml:contents rdf:type owl:ObjectProperty ;
         rdfs:comment '''Description of the contents.
         TODO: need to decide whether this is a multi-valued property with associated array coordinates or a
         single-valued property with an array value.
         Currently set to string as a "dummy" value that can serialize anything.''' ;
+        rdfs:subPropertyOf opil:compositionalProperty ;
               rdfs:domain paml:SampleArray ;
-              rdfs:range xsd:string ;
+              rdfs:range paml:PrimitiveArray ;
               rdfs:label "contents" .
 
+paml:PrimitiveArray rdf:type owl:Class ;
+        rdfs:comment '''A PrimitiveArray is a one-dimensional array of primtitive types represented as strings''' ;
+        rdfs:subClassOf sbol:Identified ,
+        [ rdf:type owl:Restriction ;
```

Comment:
This restriction is ill-formed.  It needs to be two restrictions: one for the `allValuesFrom` and one for the cardinality.

Also, do you really want the `rdfs:comment` here?  The comment will be stuck to the anonymous class of "things that have 0 or more `paml:elements` that are `uml:OrderedPropertyValue`s," (modulo the problem above) and not to the `paml:elements` property.